src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs

```diff
@@ -92,4 +92,26 @@ public override bool IsInvalid
             get { return handle == IntPtr.Zero; }
         }
     }
+
+    [SecurityCritical]
+    internal sealed class SafeSharedX509StackHandle : SafeHandle
+    {
+        internal static readonly SafeSharedX509StackHandle InvalidHandle = new SafeSharedX509StackHandle();
+
+        private SafeSharedX509StackHandle() :
+            base(IntPtr.Zero, ownsHandle: false)
+        {
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            SetHandle(IntPtr.Zero);
```

Comment:
> Maybe the creation of the "child" handle AddRef's the parent and its ReleaseHandle ReleaseRef's the parent?

Yes, this is the right patern to use for safe handles that point to container members, and where the container has to be kept alive for as long as the container member is alive.

(Windows security and networking APIs tend to have this problem too - I believe that you will find this in desktop reference sources if you look around.)